backend/plan/views.py

```diff
@@ -399,3 +406,101 @@ def get_queryset(self):
             "sections__meetings__room",
         )
         return queryset
+
+
+class CalendarAPIView(APIView):
+    schema = PcxAutoSchema(
+        response_codes={
+            reverse_func("calendar-view", args=["schedule_pk"]): {
+                "GET": {
+                    200: "Schedule exported successfully",
+                },
+            },
+        },
+    )
+
+    def get(self, *args, **kwargs):
+        """
+        Return a .ics file of the user's selected schedule
+        ---
+        responses:
+            "200":
+                description: Return a calendar file in ICS format.
+                content:
+                    text/calendar:
+                        schema:
+                            type: string
+        ---
+        """
+        schedule_pk = kwargs["schedule_pk"]
+
+        schedule = (
+            Schedule.objects.filter(pk=schedule_pk)
+            .prefetch_related("sections", "sections__meetings")
+            .first()
+        )
+
+        if not schedule:
+            return Response({"detail": "Invalid schedule"}, status=status.HTTP_403_FORBIDDEN)
+
+        day_mapping = {"M": "MO", "T": "TU", "W": "WE", "R": "TH", "F": "FR"}
```

Comment:
probably better practice to just declare this at the top level (not inside the function) and make it all caps